test/src/e2e/informer.test.ts

```diff
@@ -94,10 +94,61 @@ describe("Informer", function() {
             ) {
                 await wait(500);
             }
-            subscription.addEventListener("message", (message: any) => {
+            subscription.once("message", (message: any) => {
                 expect(message).to.be.include(JSON.stringify(message2));
             });
         });
+
+        it("De-registration", async function() {
+            const subscription = new WebSocket(
+                `ws://localhost:${nodeA.informerPort}`
+            );
+            subscription.once("open", () => {
+                subscription.send(
+                    JSON.stringify({
+                        jsonrpc: "2.0",
+                        id: 1,
+                        method: "register",
+                        params: ["PeerAdded"]
+                    })
+                );
+            });
+            let subscribtionId: string;
+            subscription.once("message", (message: string) => {
+                subscribtionId = message.substr(26, 20);
+                expect(message.includes("result")).to.be.true;
+            });
+            expect(
+                await nodeA.rpc.net.connect({
+                    address: address.toString(),
+                    port: nodeB.port
+                })
+            ).to.be.undefined;
```

Comment:
What is the meaning of checking the result with undefined?

Reply:
@MSNTCS Please reply to this comment.

Reply:
First of all, I wanted to make sure it is connected and the connection is established. Second, I just follow the same idea that we currently use in [network2.test.ts](https://github.com/CodeChain-io/foundry/blob/49035aee1af5f191ff4581837003667a070a70ee/test/src/e2e/network2.test.ts#L64)

Reply:
@MSNTCS Could you check what `to.be.undefiend` is doing?
When the `expect(...).to.be.undefined` fails?
When the `await nodeA.rpc.net.connect(...)` returns undefined ( or other thing)?

Reply:
@majecty
1- what undefined do is like when the result does not have an actual type.
2- I think in this situation it fails``` expect(1).to.be.undefined```;
3- ```await nodeA.rpc.net.connect(...)``` return void which based on our previous implementation undefined is used to make sure the connection is established.

Reply:
I think there is no point in what I did while I am checking the connectivity later.

Reply:
Thanks for researching it.
My opinion is that `expect(...connect...).to.be.undefined` doesn't do anything meaningful.
The `connect` will return undefined or throw an error. Since the function returns only undefined, the expect always passes.
If we want to assert that the `connect` does not throw, we must use `.not.throw`: https://www.chaijs.com/api/bdd/#method_throw
I checked when the `expect(xxx).to.be.undefined` came from. It was an old commit: b60982f28ee81fc50f57744b7b92a4bf7d75b682.

@MSNTCS Could you fix all unnecessary `to.be.undefined` in the later PR?

Reply:
@majecty Done

Reply:
@majecty Do you want me to change [this PR](https://github.com/CodeChain-io/foundry/commit/b60982f28ee81fc50f57744b7b92a4bf7d75b682)?

Reply:
Please create a new PR that removes `to.be.undefined` after this PR is merged.